tfm: Add TF-M interface include headers always

joerchan opened the pull request:
Add the TF-M interface include headers to the zephyr includes when building with TF-M.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
tfm: Add TF-M interface include headers always

Add the TF-M interface include headers to the zephyr includes when
building with TF-M.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan commented:
Note. This was attempted earlier and had some issues with mbedtels psa headers. Running CI on this now to discover if there are issues still.

jfischer-no reviewed (commented):
I am confused, I see `+0 -60` and read "Add TF-M interface...", please enlighten me.

joerchan commented:
> I am confused, I see `+0 -60` and read "Add TF-M interface...", please enlighten me.

Hehe, yeah I thought that was a bit funny. The important change is here:
https://github.com/zephyrproject-rtos/zephyr/pull/53097/files#diff-dc28570be9d838c291b4be7f1ea2436d81b17f095892a52869cd3cd88317c599

I added it in the zephyr include directories, and removed it from the private list of the tfm_api library.
This ended up as an all-delete diff.

joerchan commented:
> Note. This was attempted earlier and had some issues with mbedtels psa headers. Running CI on this now to discover if there are issues still.

Looking at CI there is clearly issues here still.
Will have to do more investigation on how this can be managed.

joerchan commented:
🏳️
